<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>781</width>
    <height>751</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_13">
    <item row="0" column="0">
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QGroupBox" name="groupBox_5">
       <property name="minimumSize">
        <size>
         <width>270</width>
         <height>450</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>270</width>
         <height>3000</height>
        </size>
       </property>
       <property name="title">
        <string/>
       </property>
       <layout class="QGridLayout" name="gridLayout_11">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_9">
          <item row="1" column="1">
           <widget class="QPushButton" name="pushButtonMinus">
            <property name="minimumSize">
             <size>
              <width>30</width>
              <height>30</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>30</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>15</pointsize>
             </font>
            </property>
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="pushButtonInitialize">
            <property name="minimumSize">
             <size>
              <width>170</width>
              <height>30</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>170</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Инициализиpовать</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="pushButtonPlus">
            <property name="minimumSize">
             <size>
              <width>30</width>
              <height>30</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>30</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>15</pointsize>
             </font>
            </property>
            <property name="text">
             <string>+</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0" colspan="3">
           <widget class="QTableWidget" name="tableWidget">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>250</width>
              <height>390</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>250</width>
              <height>2000</height>
             </size>
            </property>
            <property name="showGrid">
             <bool>true</bool>
            </property>
            <property name="rowCount">
             <number>5</number>
            </property>
            <property name="columnCount">
             <number>2</number>
            </property>
            <attribute name="verticalHeaderHighlightSections">
             <bool>true</bool>
            </attribute>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <column>
             <property name="text">
              <string>Ключ</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Значение</string>
             </property>
            </column>
            <item row="0" column="0">
             <property name="text">
              <string>7</string>
             </property>
            </item>
            <item row="0" column="1">
             <property name="text">
              <string>StringKey7</string>
             </property>
            </item>
            <item row="1" column="0">
             <property name="text">
              <string>4</string>
             </property>
            </item>
            <item row="1" column="1">
             <property name="text">
              <string>StringKey4</string>
             </property>
            </item>
            <item row="2" column="0">
             <property name="text">
              <string>3</string>
             </property>
            </item>
            <item row="2" column="1">
             <property name="text">
              <string>StringKey3</string>
             </property>
            </item>
            <item row="3" column="0">
             <property name="text">
              <string>5</string>
             </property>
            </item>
            <item row="3" column="1">
             <property name="text">
              <string>StringKey5</string>
             </property>
            </item>
            <item row="4" column="0">
             <property name="text">
              <string>6</string>
             </property>
            </item>
            <item row="4" column="1">
             <property name="text">
              <string>StringKey6</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="groupBox">
       <property name="minimumSize">
        <size>
         <width>270</width>
         <height>140</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>270</width>
         <height>140</height>
        </size>
       </property>
       <property name="title">
        <string>Добавление узла</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="1" column="0">
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="minimumSize">
             <size>
              <width>80</width>
              <height>30</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>80</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="text">
             <string>Ключ:</string>
            </property>
            <property name="textFormat">
             <enum>Qt::AutoText</enum>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_2">
            <property name="minimumSize">
             <size>
              <width>80</width>
              <height>30</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>80</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Значение</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QPushButton" name="pushButtonAdd">
            <property name="minimumSize">
             <size>
              <width>250</width>
              <height>30</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>250</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Добавить</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="lineEditKeyAdd">
            <property name="minimumSize">
             <size>
              <width>160</width>
              <height>30</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>160</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>10</pointsize>
             </font>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="lineEditValueAdd">
            <property name="minimumSize">
             <size>
              <width>160</width>
              <height>30</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>160</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>10</pointsize>
             </font>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="minimumSize">
        <size>
         <width>270</width>
         <height>100</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>270</width>
         <height>100</height>
        </size>
       </property>
       <property name="title">
        <string>Удаление узла</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="label_3">
            <property name="minimumSize">
             <size>
              <width>80</width>
              <height>30</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>80</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="text">
             <string>Ключ:</string>
            </property>
            <property name="textFormat">
             <enum>Qt::AutoText</enum>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="lineEditKeyRemove">
            <property name="minimumSize">
             <size>
              <width>160</width>
              <height>30</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>160</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>10</pointsize>
             </font>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QPushButton" name="pushButtonRemove">
            <property name="minimumSize">
             <size>
              <width>250</width>
              <height>30</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>250</width>
              <height>30</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Удалить</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="0" column="1">
     <layout class="QGridLayout" name="gridLayout_10">
      <item row="0" column="0" colspan="2">
       <widget class="QTreeWidget" name="treeWidget">
        <property name="minimumSize">
         <size>
          <width>480</width>
          <height>380</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>1000</width>
          <height>1000</height>
         </size>
        </property>
        <column>
         <property name="text">
          <string notr="true">1</string>
         </property>
        </column>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QSplitter" name="splitter_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <widget class="QGroupBox" name="groupBox_6">
         <property name="minimumSize">
          <size>
           <width>270</width>
           <height>90</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>270</width>
           <height>90</height>
          </size>
         </property>
         <property name="title">
          <string/>
         </property>
         <layout class="QGridLayout" name="gridLayout_12">
          <item row="0" column="0">
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QPushButton" name="pushButtonBalance">
              <property name="minimumSize">
               <size>
                <width>250</width>
                <height>30</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>250</width>
                <height>30</height>
               </size>
              </property>
              <property name="font">
               <font>
                <pointsize>10</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Сбалансировать</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pushButtonShow">
              <property name="minimumSize">
               <size>
                <width>250</width>
                <height>30</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>250</width>
                <height>30</height>
               </size>
              </property>
              <property name="font">
               <font>
                <pointsize>10</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Вывести элементы</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="minimumSize">
          <size>
           <width>270</width>
           <height>140</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>270</width>
           <height>140</height>
          </size>
         </property>
         <property name="title">
          <string>Поиск</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="0" column="0">
           <layout class="QGridLayout" name="gridLayout_5">
            <item row="0" column="0">
             <widget class="QLabel" name="label_7">
              <property name="minimumSize">
               <size>
                <width>80</width>
                <height>30</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>80</width>
                <height>30</height>
               </size>
              </property>
              <property name="font">
               <font>
                <pointsize>10</pointsize>
               </font>
              </property>
              <property name="layoutDirection">
               <enum>Qt::LeftToRight</enum>
              </property>
              <property name="text">
               <string>Ключ:</string>
              </property>
              <property name="textFormat">
               <enum>Qt::AutoText</enum>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="lineEditSearch">
              <property name="minimumSize">
               <size>
                <width>160</width>
                <height>30</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>160</width>
                <height>30</height>
               </size>
              </property>
              <property name="font">
               <font>
                <pointsize>10</pointsize>
               </font>
              </property>
             </widget>
            </item>
            <item row="1" column="0" colspan="2">
             <widget class="QPushButton" name="pushButtonSearch">
              <property name="minimumSize">
               <size>
                <width>250</width>
                <height>30</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>250</width>
                <height>30</height>
               </size>
              </property>
              <property name="font">
               <font>
                <pointsize>10</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Найти</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0" colspan="2">
             <widget class="QLabel" name="labelValueSearch">
              <property name="minimumSize">
               <size>
                <width>250</width>
                <height>30</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>250</width>
                <height>30</height>
               </size>
              </property>
              <property name="font">
               <font>
                <pointsize>10</pointsize>
               </font>
              </property>
              <property name="layoutDirection">
               <enum>Qt::LeftToRight</enum>
              </property>
              <property name="text">
               <string>Значение:</string>
              </property>
              <property name="textFormat">
               <enum>Qt::AutoText</enum>
              </property>
              <property name="alignment">
               <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QGroupBox" name="groupBox_4">
         <property name="minimumSize">
          <size>
           <width>270</width>
           <height>100</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>270</width>
           <height>100</height>
          </size>
         </property>
         <property name="title">
          <string>Поменять информацию местами</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_8">
          <item row="0" column="0">
           <layout class="QGridLayout" name="gridLayout_7">
            <item row="0" column="0" colspan="2">
             <widget class="QPushButton" name="pushButtonSwap">
              <property name="minimumSize">
               <size>
                <width>250</width>
                <height>30</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>250</width>
                <height>30</height>
               </size>
              </property>
              <property name="font">
               <font>
                <pointsize>10</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Свапнуть</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QTextEdit" name="textEdit">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>340</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>500</width>
          <height>500</height>
         </size>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
